File: ios/loveaceios/Models/DoorCardModels.swift
import Foundation

struct DoorCardUserInfo: Codable {
    let personId: String
    let personName: String
    let cardId: String
    let personKind: Int

    enum CodingKeys: String, CodingKey {
        case personId = "PersonID"
        case personName = "PersonName"
        case cardId = "CardID"
        case personKind = "PersonKind"
    }

    init(personId: String = "", personName: String = "", cardId: String = "", personKind: Int = 0) {
        self.personId = personId; self.personName = personName
        self.cardId = cardId; self.personKind = personKind
    }
}

struct DoorCardRoom: Codable, Identifiable {
    var id: String { roomId }
    let roomId: String
    let roomName: String
    let buildName: String
    let btMac: String
    let sKey: String
    let sn: Int
    let power: Int
    let endDateTime: String
    let personId: String
    let schoolId: String

    enum CodingKeys: String, CodingKey {
        case roomId = "RoomID"
        case roomName = "RoomName"
        case buildName = "BuildName"
        case btMac = "BtMac"
        case sKey
        case sn = "SN"
        case power = "Power"
        case endDateTime = "EndDateTime"
        case personId = "PersonID"
        case schoolId = "SchoolID"
    }

    init(roomId: String = "", roomName: String = "", buildName: String = "",
         btMac: String = "", sKey: String = "", sn: Int = 0, power: Int = 0,
         endDateTime: String = "", personId: String = "", schoolId: String = "") {
        self.roomId = roomId; self.roomName = roomName; self.buildName = buildName
        self.btMac = btMac; self.sKey = sKey; self.sn = sn; self.power = power
        self.endDateTime = endDateTime; self.personId = personId; self.schoolId = schoolId
    }
}

struct DoorCardCredentials: Codable {
    let username: String
    let userno: String
    let password: String

    init(username: String = "", userno: String = "", password: String = "") {
        self.username = username; self.userno = userno; self.password = password
    }
}

enum BleConnectionState {
    case disconnected
    case scanning
    case connecting
    case connected
    case error
}

enum DoorOperation: CaseIterable {
    case openDoor
    case addCard
    case freezeCard
    case checkTime
    case alwaysOpen
    case alwaysOff
    case checkDaily

    var label: String {
        switch self {
        case .openDoor: return "手机开门"
        case .addCard: return "发卡"
        case .freezeCard: return "冻结卡"
        case .checkTime: return "校时"
        case .alwaysOpen: return "常开设置"
        case .alwaysOff: return "常闭设置"
        case .checkDaily: return "考勤"
        }
    }

    var systemImage: String {
        switch self {
        case .openDoor: return "lock.open.fill"
        case .addCard: return "creditcard.fill"
        case .freezeCard: return "snowflake"
        case .checkTime: return "clock.fill"
        case .alwaysOpen: return "lock.open"
        case .alwaysOff: return "lock.fill"
        case .checkDaily: return "checkmark.circle.fill"
        }
    }
}

File: android/app/src/main/java/tech/loveace/appv3/data/model/DoorCardModels.kt
package tech.loveace.appv3.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// ==================== Door Card Models ====================

/** 门卡系统用户信息（登录后返回） */
@Serializable
data class DoorCardUserInfo(
    @SerialName("PersonID") val personId: String = "",
    @SerialName("PersonName") val personName: String = "",
    @SerialName("CardID") val cardId: String = "",
    @SerialName("PersonKind") val personKind: Int = 0, // 0=学生, 1=管理员
)

/** 房间/门锁信息 */
@Serializable
data class DoorCardRoom(
    @SerialName("RoomID") val roomId: String = "",
    @SerialName("RoomName") val roomName: String = "",
    @SerialName("BuildName") val buildName: String = "",
    @SerialName("BtMac") val btMac: String = "",
    @SerialName("sKey") val sKey: String = "",
    @SerialName("SN") val sn: Int = 0,
    @SerialName("Power") val power: Int = 0,
    @SerialName("EndDateTime") val endDateTime: String = "",
    @SerialName("PersonID") val personId: String = "",
    @SerialName("SchoolID") val schoolId: String = "",
)

/** 门卡绑定凭证 */
@Serializable
data class DoorCardCredentials(
    val username: String = "",  // 姓名
    val userno: String = "",    // 学号
    val password: String = "",  // MD5 密码
)

/** BLE 连接状态 */
enum class BleConnectionState {
    Disconnected,
    Scanning,
    Connecting,
    Connected,
    Error,
}

/** 门锁操作类型 */
enum class DoorOperation(val label: String, val icon: String) {
    OpenDoor("手机开门", "🔓"),
    AddCard("发卡", "💳"),
    FreezeCard("冻结卡", "🧊"),
    CheckTime("校时", "⏰"),
    AlwaysOpen("常开设置", "🔓"),
    AlwaysOff("常闭设置", "🔒"),
    CheckDaily("考勤", "📋"),
}
